File: CS/PrismOnDXDocking.Infrastructure/Adapters/LayoutPanelAdapter.cs
using System.ComponentModel.Composition;
using System.Windows.Controls;
using DevExpress.Xpf.Docking;
using System;
using Prism.Regions;

namespace PrismOnDXDocking.Infrastructure.Adapters {
    [Export(typeof(LayoutPanelAdapter)), PartCreationPolicy(CreationPolicy.NonShared)]
	public class LayoutPanelAdapter : RegionAdapterBase<LayoutPanel> {
        [ImportingConstructor]
		public LayoutPanelAdapter(IRegionBehaviorFactory behaviorFactory) : 
            base(behaviorFactory) { 
		}
		protected override IRegion CreateRegion() {
			return new SingleActiveRegion();
		}
		protected override void Adapt(IRegion region, LayoutPanel regionTarget) {
			string regionName = RegionManager.GetRegionName(regionTarget);
			if(!String.IsNullOrEmpty(regionName)) {
				if(regionTarget != null && regionTarget.Content == null) {
					ContentControl control = new ContentControl();
					RegionManager.SetRegionName(control, regionName);
                    regionTarget.Content = control;
				}
			}
		}
	}
}

File: VB/PrismOnDXDocking.Infrastructure/Adapters/LayoutPanelAdapter.vb
Imports System.ComponentModel.Composition
Imports System.Windows.Controls
Imports DevExpress.Xpf.Docking
Imports Prism.Regions

Namespace PrismOnDXDocking.Infrastructure.Adapters
    <Export(GetType(LayoutPanelAdapter)), PartCreationPolicy(CreationPolicy.NonShared)>
    Public Class LayoutPanelAdapter
        Inherits RegionAdapterBase(Of LayoutPanel)
        <ImportingConstructor>
        Public Sub New(ByVal BehaviorFactory As IRegionBehaviorFactory)
            MyBase.New(BehaviorFactory)
        End Sub
        Protected Overrides Function CreateRegion() As IRegion
            Return New SingleActiveRegion()
        End Function
        Protected Overrides Sub Adapt(ByVal region As IRegion, ByVal regionTarget As LayoutPanel)
            Dim regionName As String = RegionManager.GetRegionName(regionTarget)
            If (Not String.IsNullOrEmpty(regionName)) Then
                If regionTarget IsNot Nothing AndAlso regionTarget.Content Is Nothing Then
                    Dim control As New ContentControl()
                    RegionManager.SetRegionName(control, regionName)
                    regionTarget.Content = control
                End If
            End If
        End Sub
    End Class
End Namespace
